src/cpu/core_dyn_x86/risc_x64.h

```diff
@@ -1149,20 +1150,22 @@ static Bit8u * gen_create_branch_long(BranchTypes type) {
 	return (cache.pos-4);
 }
 
-static void gen_fill_branch_long(Bit8u * data,Bit8u * from=cache.pos) {
-	*(Bit32u*)data=(Bit32u)(from-data-4);
+static void gen_fill_branch_long(uint8_t *data, uint8_t *from = cache.pos)
+{
+	host_writed(data, from - data - sizeof(uint32_t));
 }
 
-static Bit8u * gen_create_jump(Bit8u * to=0) {
+static uint8_t *gen_create_jump(uint8_t *to = 0)
+{
 	/* First free all registers */
 	cache_addb(0xe9);
-	cache_addd((Bit32u)(to-(cache.pos+4)));
-	return (cache.pos-4);
+	cache_addd(to - cache.pos - sizeof(uint32_t));
+	return cache.pos - sizeof(uint32_t);
 }
 
 #if 0
-static void gen_fill_jump(Bit8u * data,Bit8u * to=cache.pos) {
-	*(Bit32u*)data=(Bit32u)(to-data-4);
```

Comment:
PPC: Fix needs to be aligned and native endian (not swapped to little-endian like the host_* functions do).